Sources/Lingo/DataSources/FileDataSource.swift

```diff
@@ -69,32 +59,30 @@ fileprivate extension FileDataSource {
     ///   "one": "You have an unread message."
     ///   "many": "You have %{count} unread messages."
     /// }
-    func pluralizedValues(fromRaw rawPluralizedValues: [String: String]) -> [PluralCategory: String] {
-        var pluralizedValues = [PluralCategory: String]()
+    func pluralizedValues(fromRaw rawPluralizedValues: [String: String]) throws -> [PluralCategory: String] {
+        var result = [PluralCategory: String]()
         
         for (rawPluralCategory, value) in rawPluralizedValues {
             guard let pluralCategory = PluralCategory(rawValue: rawPluralCategory) else {
-                assertionFailure("Unsupported plural category: \(rawPluralCategory)")
-                continue
+                throw Error.parsingFailure(message: "Unsupported plural category: \(rawPluralCategory)")
             }
-            pluralizedValues[pluralCategory] = value
+            result[pluralCategory] = value
         }
         
-        return pluralizedValues
+        return result
```

Comment:
If the result is empty, maybe throwing here instead of returning it empty?

Reply:
I don't think it is necessary to throw here, as the function only does a type conversion, so if the input is empty, so the output would be, but what matters is that it will be of a correct type.